blog.js

```diff
@@ -9,10 +9,51 @@ rawData.sort(function (a, b) {
 });
 
 var blogConstructor = function () {
+  var authors = [];
+  var categories = [];
+  // articleConstructor(rawData[0], 0);
   for (var i = 0; i < rawData.length; i++) {
+    authors.push(rawData[i].author);
+    categories.push(rawData[i].category);
     articleConstructor(rawData[i], i);
   };
+  $('article section p:not(:first-child)').addClass('hide').hide();
   $('article:first').remove();
+  authors = $.unique(authors);
+  categories = $.unique(categories);
+  for (var i = 0; i < authors.length; i++) {
+    $('#authors').append('<option value='+authors[i].trim()+'>'+authors[i]+'</option>');
```

Comment:
Using .trim() causes a bug that's breaking your page. Your option values are only adding the first word of a name and then putting invalid html into the DOM:

![screen shot 2015-12-04 at 3 09 51 pm](https://cloud.githubusercontent.com/assets/3937302/11603735/2b55ba04-9a99-11e5-9f2a-47daa17c1d39.png)

Notice `<option value="Betty" crocker="">Betty Crocker</option>` is only assigning "Betty" to value, and then hanging "crocker" into the option tag? I'm guessing you are trying to use `.trim()` to remove the whitespace in the string and then end up with "BettyCrocker"? What you need to use is `.replace()`, so something like `"hello world".replace(" ", "");` will result in `:"helloworld"`

Also, you should consider just changing attributes in your HTML instead of directly injecting html. http://api.jquery.com/attr/ is a good way to do that.